### DISCUSSION TITLE
Welcome to writingtool Discussions!

### DISCUSSION THREAD

**Discussion by FredKruse** [writingtool-org/writingtool#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by lobaluna** [comment]
Hi, I just discovered this extension. I tried it and I like it.

Congratulations and thank you very much!

As for further enhance this tool, I propose to add a contextual menu where every correction gets corrected: 1) everywhere in the document, 2) in selected text/area, and 3) just the single instance of error word.

**Comment by aldQueiroz** [comment]
Hi,
I've been facing issues to make WritingTool extension working as meant within my LibreOffice installations and, apparently, all of them would be related to WritingTool **dependency of Java Runtime**. Rather than discussing how to solve the current issues, I'd like to seek out how to port WritingTool extension to another programming language, e.g. Python, in order to avoid Java dependency.

**Reply by lobaluna** [reply]
Good decision. I'm sorry I cannot program that good, but I'll be glad to help testing it

**Reply by FredKruse** [reply]
The idea is fundamentally good, but unfortunately, it won't work. WritingTool is based on LanguageTool. The underlying check functions are the same. LanguageTool is written in Java and requires version 17. If you wanted to port WritingTool to Python, you would also have to port the entire LanguageTool to Python.

**Reply by lobaluna** [reply]
Nevertheless, thanks for all your effort.

**Comment by VickyG2026** [comment]
Hello.  I love the product but spent over 6 hours resolving my Java issues. Google is not your friend when trying to figure out the cryptic java error messages from inside LibreOffice.  

A lot of my issues could have been avoided if this sentence on your extension page was more complete.  
**Important Note**: WritingTool require **Java version 17** or higher now

Suggested
**Important Note**: WritingTool requires **Java version 17** or higher.   Before you try to enable WT,  navigate to **Tools->Options->LibreOffice->Advanced** and make sure "Use a java runtime environment"  is checked and that the Java version is 17 or higher is selected.  If you don't have a compatible version of Java, download and install a JDK for a stable version of Java from Oracle.com.  Make sure to choose a version compatible with your operating system.

**Comment by VickyG2026** [comment]
Oops, grammar correction, that should have said. 

Important Note: WritingTool requires **Java version 17** or higher. Before you try to enable WT, navigate to **Tools->Options->LibreOffice->Advanced** and make sure **"Use a java runtime environment"** is checked and that the Java version selected is 17 or higher. If you don't have a compatible version of Java, download and install a JDK for a stable version of Java from Oracle.com. Make sure to choose a version compatible with your operating system.

Even these instructions are on the light side for what most of my friends could follow. 
I am somewhat tech savvy but was stymied by the WT intallation.  Google results had me extracting the JRE from the JDK, changing environmental variables, diving into PowerShell and administrative command prompts. . . .  Doing everything but selecting the Java version in LibreOffice.  I had no idea that there even was a setting there.

**Reply by FredKruse** [reply]
Thank you for the suggestion. I've just added it to the LibreOffice Extensions page. It's now awaiting publication.
